From a0fcf4f45a88180ea5e5a2f71aa27f7ab4604bf3 Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@hadrons.org>
Date: Fri, 16 Aug 2019 04:34:37 +0200
Subject: [PATCH libaio] harness: Print better error messages on error
 conditions

These should help diagnose problems when dealing with error failures.

Signed-off-by: Guillem Jover <guillem@hadrons.org>
---
 harness/cases/22.t | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/harness/cases/22.t b/harness/cases/22.t
index b13024e..8014dcf 100644
--- a/harness/cases/22.t
+++ b/harness/cases/22.t
@@ -76,7 +76,7 @@ int test_main(void)
 
 		ret = io_setup(1, &ctx);
 		if (ret) {
-			printf("child: io_setup failed\n");
+			printf("child: io_setup failed: %s\n", strerror(-ret));
 			return 1;
 		}
 
@@ -86,7 +86,7 @@ int test_main(void)
 			/* if poll isn't supported, skip the test */
 			if (ret == -EINVAL)
 				return 3;
-			printf("child: io_submit failed\n");
+			printf("child: io_submit failed: %s\n", strerror(-ret));
 			return 1;
 		}
 
@@ -99,7 +99,7 @@ int test_main(void)
 		} while (ret == 0);
 
 		if (ret != -EINTR) {
-			printf("child: io_pgetevents did not set errno to EINTR\n");
+			printf("child: io_pgetevents did not set errno to EINTR: %s\n", strerror(-ret));
 			return 1;
 		}
 
@@ -117,7 +117,7 @@ int test_main(void)
 
 		ret = io_setup(1, &ctx);
 		if (ret) {
-			printf("parent: io_setup failed\n");
+			printf("parent: io_setup failed: %s\n", strerror(-ret));
 			return 1;
 		}
 
@@ -126,7 +126,7 @@ int test_main(void)
 			/* if poll isn't supported, skip the test */
 			if (ret == -EINVAL)
 				return 3;
-			printf("parent: io_submit failed with %d\n", ret);
+			printf("parent: io_submit failed with %d: %s\n", ret, strerror(-ret));
 			return 1;
 		}
 
@@ -134,7 +134,7 @@ int test_main(void)
 
 		ret = io_pgetevents(ctx, 1, 1, &ev, NULL, &sigmask);
 		if (ret < 0) {
-			printf("parent: io_pgetevents failed\n");
+			printf("parent: io_pgetevents failed: %s\n", strerror(-ret));
 			return 1;
 		}
 		if (ret != 1) {
-- 
2.23.0.rc1.170.gbd704faa3e

